### PR TITLE
Use strict comparision

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Closes #

Changes proposed in this pull request:
Use strict comparison

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

File: app/Controllers/subscriptionController.php [hunk 1]
```diff
@@ -10,7 +10,7 @@ class FreshRSS_subscription_Controller extends Minz_ActionController {
 	 * underlying framework.
 	 */
 	public function firstAction() {
-		if (!FreshRSS_Auth::hasAccess()) {
+		if (false !== FreshRSS_Auth::hasAccess()) {
```
Comment (Frenzie) [inline app/Controllers/subscriptionController.php:13]: Setting aside that this is incorrect (`=== false` or `!== true`), should this be strict at all? It seems to me you want it to fail as much as possible, i.e., when falsy, not when strictly false.

**Comment by ColonelMoutarde** [comment]
"Prior to PHP 8.0.0, if a string is compared to a number or a numeric string then the string was converted to a number before performing the comparison. This can lead to surprising results... "

https://www.php.net/manual/en/language.operators.comparison.php

**Comment by Frenzie** [comment]
Sorry, but I don't understand what you're getting at. The function in question always returns a boolean.

**Comment by Frenzie** [comment]
If we grant for the sake of argument that we should always be using strict comparisons, then we should be using `FreshRSS_Auth::hasAccess() === true` and `FreshRSS_Auth::hasAccess() !== true`. NB I don't think this is something we should be doing.

Using `=== false` or `!== false` would make a (however unlikely) falsy value mean that the unauthorized user is treated as being logged in.

In any case, `!FreshRSS_Auth::hasAccess()` is the same thing as `FreshRSS_Auth::hasAccess() !== true` (or rather it's the same as `FreshRSS_Auth::hasAccess() != true`, but that's not a meaningful distinction in this context).

**Comment by Alkarex** [comment]
There is most likely quite many places in the code to improve, but this specific original line looks fine for now, as Frenzie explained :-)

**Pull request closed without merging**